content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,116 @@
+We thus do not have to download or reference any libraries in the `Makefile` and can leave the `LIBS` variable empty.
+
+With the `Makefile` present and the directory structure correctly set up, we are able to configure the unikernel as the next step.
+This will prompt a text-based user interface that will allow us to select the target architecture (e.g., `aarch64` or `x86-64`) and platform (e.g., `KVM` or `Xen`).
+
+```console
+$ make menuconfig
+```
+
+By default the build tool selects `x86-64` as target architecture, so we only have to choose a target platform.
+We will decide for `KVM` by navigating to the `Platform Configuration | KVM guest` option and selecting it.
+Afterwards, we save and exit the configuration menu.
+
+You can then build the unikernel with:
+
+```console
+$ make
+```
+
+Consider adding `-jX` with `X` being the number of CPUs in your machine to speed up the build process.
+
+When the build completes you will have a Unikraft-based kernel image (`build/app-helloworld_kvm-x86_64`) that you can start with `QEMU/KVM` using the `-kernel` option.
+See [here](docs/operations/plats/kvm/#build-and-run-unikraft-on-kvm-using-make-and-qemu) for further information on running the unikernel with `KVM`.
+
+## Adding external dependencies
+
+When the application becomes more complex, it is likely that not all functionality is included in the Unikraft core repository.
+As you might have noticed already from the code organization in the core repository (`lib/*`), Unikraft puts an emphasis on encapsulating functionality in libraries that can be easily added, removed, or replaced to tailor the unikernel to the application's needs.
+
+To add a library you simply clone it into the `libs` directory created earlier.
+In this example, we add the `lightweight IP` (`lwip`) network stack [`lib-lwip`](https://github.com/unikraft/lib-lwip):
+
+```console
+$ cd my-unikernel
+$ cd libs
+$ git clone https://github.com/unikraft/lib-lwip
+```
+
+Next, we have reference the library in the `Makefile` of `app-helloworld` so that it becomes available for selection in the configuration:
+
+```Makefile
+UK_LIBS ?= $(PWD)/../../libs
+
+LIBS := $(UK_LIBS)/lib-lwip
+#LIBS := $(LIBS-y):$(UK_LIBS)/lib-X
+```
+
+We can now re-run
+
+```bash
+make menuconfig
+```
+
+and select `lwip` in the `Library Configuration` sub-menu.
+We can then rebuild the unikernel, by running:
+
+```bash
+$ make clean
+$ make prepare
+$ make
+```
+
+The `prepare` step gives the build tool the chance to pull the actual source code for `lwip` and apply necessary patches to make it work with Unikraft.
+This has to be done only once after adding a new library.
+
+Note that the build may fail because depending on the configuration, `lwip` requires a more extensive `libc` (e.g. [`musl`](https://github.com/unikraft/lib-musl)).
+You can add these additional dependencies just as described.
```

Comment:
Add a section titled `## Example: Using Make with app-helloworld-cpp` and describe the steps into getting `app-helloworld-cpp` working with Make. Add `Makefile`, `Makefile.uk`, `Config.uk` and get things going.